test/test_model.py

```diff
@@ -3791,5 +3791,82 @@ def test_repr(self):
         )
 
 
+class TestCloudCredential(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cloud_credential_dict = {
```

Comment:
I think this is better just as a local variable inside `test_from_dict`, as it reduces the scope and it's only used by that test. Could probably shorten the name to `dct` or `d` in that case. Let's also include two tests: one with only the required fields, and one with all fields (including `redacted`).